tests/bayesian/experiment_suite.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/python2.7
+import random
+import matplotlib
+matplotlib.use("Agg")
+import warnings
+import json
+import os
+
+from htmresearch.frameworks.layers.object_machine_factory import (createObjectMachine)
+from htmresearch.frameworks.layers.l2_l4_inference import L4L2Experiment
+
+
+def runExperiments():
+    # default parameters
+    seed = 40
+    suppressWarnings = True
+    maxNumSegments = 2
+    sdrSize = 40
+    columns_count = 2048
+    cells_per_column = 8
+
+    # parameters to iterate over
+    num_columns_p = [1, 3]
+    implementation_p = ["SummingBayesian", "Bayesian"]
+    learning_rate_l2_p = [0.1, 0.2]
+    learning_rate_l4_p = [0.1, 0.2]
+    activation_threshold_p = [0.3, 0.5, 0.7]
+    min_threshold_p = [0.1, 0.3, 0.5]
+    forgetting_p = [0.01, 0.1, 0.15]
+    use_support_p = [True, False]
+    use_proximal_probabilities_p = [True, False]
+    use_apical_tiebreak_p = [True, False]
+
+    for num_columns in num_columns_p:
```

Comment:
please check out itertools. You can find an example in this implementation https://github.com/leoTiez/htmresearch/blob/experiment/grid-search-5-objects/experiments.py